### PR TITLE
Server: Database layer

### PR DESCRIPTION
**Pull request by promaty**
Merge after #3

### CODE REVIEW

File: packages/shared/src/database/migrations/20200716154154_create-files.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  return knex.schema.createTable('Files', function (table) {
+    table.increments('id')
+    table.string('owner').index().notNullable()
+    table.string('cid').unique().notNullable()
+    table.boolean('verified').index().defaultTo(false).notNullable()
+    table.integer('sizeBytes').unsigned().notNullable()
+    table.string('originalName')
+    table.string('encoding')
+    table.string('mimeType')
+    table.dateTime('expiresAt').index()
```
Comment (facuspagnuolo) [inline packages/shared/src/database/migrations/20200716154154_create-files.ts:14]: This one is for the authentication right?

Comment (promaty) [reply packages/shared/src/database/migrations/20200716154154_create-files.ts:14]: Yep I will run a worker that will delete unverified files after this date

File: packages/server/src/controllers/files.ts [hunk 2]
```diff
@@ -2,59 +2,65 @@ import { Middleware } from 'koa'
...
     }
+    ctx.body = await query
   }
 
   static findOne: Middleware = async (ctx) => {
-    FilesValidator.validateForGetSingle(ctx)
-    ctx.body = dummy_file
+    await FilesValidator.validateForFindOne(ctx)
+    const { cid } = ctx.params
+    ctx.body = await File.findOne({cid})
```
[the remark this answers is not in the document]
Comment (promaty) [reply packages/server/src/controllers/files.ts:64]: good point will update

File: packages/server/src/controllers/files.ts [hunk 3]
```diff
@@ -2,58 +2,64 @@ import { Middleware } from 'koa'
...
+        mimeType: file.mimetype
+      })
     }
     catch (err) {
       await deleteTempFile(file.path)
       throw err
     }
     await deleteTempFile(file.path)
-    ctx.body = dummy_file
+    ctx.body = await File.findOne({cid})
```
Comment (facuspagnuolo) [inline packages/server/src/controllers/files.ts:39]: Shouldn't these two lines above be inside the `try` clause?

Comment (promaty) [reply packages/server/src/controllers/files.ts:39]: I just need that try catch to make sure I delete the temp file, I don't have it at this point. There is still general error handler as top middleware.

**Pull request merged**